### PR TITLE
user-data secret is created with the key "https://github.com/openshif…

### PR DESCRIPTION
**Pull request by raz-bn**
The issue explain the PR, this issue will be closed after fixes in the cluster-api-provider-kubevirt

**What this PR does / why we need it**: userdata secret is now created with the key userdata instead of value

**Which issue(s) this PR fixes** #1634

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -807,7 +807,7 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
 	}
 	userDataSecret.Data = map[string][]byte{
 		"disableTemplating": []byte(base64.StdEncoding.EncodeToString([]byte("true"))),
-		"value":             userDataValue,
+		"userdata":          userDataValue,
```
Comment (alvaroaleman) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:810]: I don't understand how this is supposed to change a key from an url?

**Comment by enxebre** [comment]
I see no reason for us to change this. It's indeed part of the contract with providers https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7e51e45a1f1a150dc63c8f02290760eb05675ec5/azure/scope/machine.go#L626
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/scope/machine.go#L280

Please, let's clarify and discuss in kubevirt.

**Comment by raz-bn** [comment]
> I see no reason for us to change this. It's indeed part of the contract with providers https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7e51e45a1f1a150dc63c8f02290760eb05675ec5/azure/scope/machine.go#L626 https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/scope/machine.go#L280
> 
> Please, let's clarify and discuss in kubevirt.

After inspecting the other providers I agree.
But every provider should be aware of the secret rotation, I do not know about the other providers but the kubevirt provider does not.
I will work on implementing such mechanism.

**Comment by enxebre** [comment]
>But every provider should be aware of the secret rotation, I do not know about the other providers but the kubevirt provider does not.
I will work on implementing such mechanism on the cluster-api-provider-kubevirt project.

I'm not sure I follow. Token rotation is driven transparently by hypershift. Userdata secret is consumed by providers via infra template. I wouldn't expect a provider needing to be aware of rotation at all.
Please point me to kubevirt issue/PR so I'd be happy to share feedback there.

**Comment by raz-bn** [comment]
> >But every provider should be aware of the secret rotation, I do not know about the other providers but the kubevirt provider does not.
> 
> I will work on implementing such mechanism on the cluster-api-provider-kubevirt project.
> 
> 
> 
> @raz-bn I'm not sure I follow. Token rotation is driven transparently by hypershift. Userdata secret is consumed by providers via infra template. I wouldn't expect a provider needing to be aware of rotation at all.
> 
> Please point me to kubevirt issue/PR so I'd be happy to share feedback there.

https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/issues/169

Please share your thoughts 😊

**Comment by davidvossel** [comment]
yes, This is an issue with the cluster-api-provider-kubevirt. That controller is mismanaging the user data secret

**Pull request closed without merging**